analyzers/tests/SonarAnalyzer.Test/Extensions/ISymbolExtensionsTest.cs

```diff
@@ -222,6 +281,125 @@ public void IsPrimaryConstructor_CS(string code, bool hasPrimaryConstructor)
+        ISymbolExtensionsCommon.AnyAttributeDerivesFromAny(null, []).Should().BeFalse();
+
+    [TestMethod]
+    public void GetAttributesForKnownType_WhenSymbolIsNull_ReturnsEmpty() =>
+        ISymbolExtensionsCommon.GetAttributes(null, KnownType.Void).Should().BeEmpty();
+
+    [TestMethod]
+    public void GetAttributesForKnownTypes_WhenSymbolIsNull_ReturnsEmpty() =>
+        ISymbolExtensionsCommon.GetAttributes(null, ImmutableArray.Create(KnownType.Void)).Should().BeEmpty();
+
+    [TestMethod]
+    public void GetParameters_WhenSymbolIsNotMethodOrProperty_ReturnsEmpty() =>
+        ISymbolExtensionsCommon.GetParameters(Mock.Of<ISymbol>(x => x.Kind == SymbolKind.Alias)).Should().BeEmpty();
+
+    [TestMethod]
+    public void GetInterfaceMember_WhenSymbolIsNull_ReturnsEmpty() =>
+        ISymbolExtensionsCommon.GetInterfaceMember((ISymbol)null).Should().BeNull();
+
+    [TestMethod]
+    public void GetOverriddenMember_WhenSymbolIsNull_ReturnsEmpty() =>
+        ISymbolExtensionsCommon.GetOverriddenMember((ISymbol)null).Should().BeNull();
+
+    [TestMethod]
+    public void GetEffectiveAccessibility_WhenSymbolIsNull_ReturnsNotApplicable() =>
+        ISymbolExtensionsCommon.GetEffectiveAccessibility(null).Should().Be(Accessibility.NotApplicable);
+
+    [DataTestMethod]
+    [DataRow(SymbolKind.Alias, "alias")]
+    [DataRow(SymbolKind.ArrayType, "array")]
+    [DataRow(SymbolKind.Assembly, "assembly")]
+    [DataRow(SymbolKind.Discard, "discard")]
+    [DataRow(SymbolKind.DynamicType, "dynamic")]
+    [DataRow(SymbolKind.ErrorType, "error")]
+    [DataRow(SymbolKind.Event, "event")]
+    [DataRow(SymbolKind.Field, "field")]
+    [DataRow(SymbolKind.FunctionPointerType, "function pointer")]
+    [DataRow(SymbolKind.Label, "label")]
+    [DataRow(SymbolKind.Local, "local")]
+    [DataRow(SymbolKind.Namespace, "namespace")]
+    [DataRow(SymbolKind.NetModule, "netmodule")]
+    [DataRow(SymbolKind.Parameter, "parameter")]
+    [DataRow(SymbolKind.PointerType, "pointer")]
+    [DataRow(SymbolKind.Preprocessing, "preprocessing")]
+    [DataRow(SymbolKind.Property, "property")]
+    [DataRow(SymbolKind.RangeVariable, "range variable")]
+    [DataRow(SymbolKind.TypeParameter, "type parameter")]
+    public void GetClassification_SimpleKinds(SymbolKind symbolKind, string expected)
+    {
+        var fakeSymbol = new Mock<ISymbol>();
+        fakeSymbol.Setup(x => x.Kind).Returns(symbolKind);
+        ISymbolExtensionsCommon.GetClassification(fakeSymbol.Object).Should().Be(expected);
+    }
+
+    [TestMethod]
+    public void GetClassification_UnknowKind()
+    {
+        var fakeSymbol = new Mock<ISymbol>();
+        fakeSymbol.Setup(x => x.Kind).Returns((SymbolKind)999);
+#if DEBUG
+        new Action(() => ISymbolExtensionsCommon.GetClassification(fakeSymbol.Object)).Should().Throw<NotSupportedException>();
+#else
+        ISymbolExtensionsCommon.GetClassification(fakeSymbol.Object).Should().Be("symbol");
+#endif
```

Comment:
The implementation behaves differently under debug and release builds

Reply:
I mean, if we run UTs in both configurations (Debug & Release), it is fine. Otherwise, I don't see the point to have these.
Do we run UTs in both configurations in the CI?

Reply:
We run release in CI and debug locally